pkg/resources/server_test.go

```diff
@@ -113,24 +119,56 @@ var _ = Describe("Server", func() {
 				err := rs.Start()
 				Expect(err).NotTo(HaveOccurred())
 
+				// channel should receive after calling rs.restart()
+				go Eventually(rs.termSignal, time.Second*5).Should(Receive())
+
+				// wait 0.1sec to ensure that we started listening in the Eventually assertion above
+				time.Sleep(100 * time.Millisecond)
 				err = rs.restart()
 				Expect(err).NotTo(HaveOccurred())
 
-				go func() {
-					rs.Watch()
-				}()
+				// both channels should receive after calling rs.Stop()
+				go Eventually(rs.termSignal, time.Second*5).Should(Receive())
+				go Eventually(rs.stopWatcher, time.Second*5).Should(Receive())
```

Comment:
ditto. 
Please move these 2 lines after the line `err = rs.Stop()`
the go threads won't be needed.

Reply:
moving them causes deadlock, please look at this example  https://play.golang.org/p/5GOt76wjGVr (deadlock)
and this one as well https://play.golang.org/p/J9TXeG_3yHD (goroutine)

also, when channels are not consumed it also causes a deadlock so I can't get rid of those `Eventually().Should(Receive())` assertions

to be honest I can't think of anything else than using goroutines for this